x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/support/StringMatcherTests.java

```diff
@@ -0,0 +1,128 @@
+            assertMatch(matcher, randomAlphaOfLength(1) + notStr);
+            assertMatch(matcher, notStr + randomAlphaOfLength(1));
+        }
+
+    }
+
+    public void testMultiplePatterns() throws Exception {
+        final String prefix1 = randomAlphaOfLengthBetween(3, 5);
+        final String prefix2 = randomAlphaOfLengthBetween(5, 8);
+        final String prefix3 = randomAlphaOfLengthBetween(10, 12);
+        final String suffix1 = randomAlphaOfLengthBetween(5, 10);
+        final String suffix2 = randomAlphaOfLengthBetween(8, 12);
+        final String exact1 = randomValueOtherThanMany(
+            s -> s.startsWith(prefix1) || s.startsWith(prefix2) || s.startsWith(prefix3) || s.endsWith(suffix1) || s.endsWith(suffix2),
+            () -> randomAlphaOfLengthBetween(5, 9));
+        final String exact2 = randomValueOtherThanMany(
+            s -> s.startsWith(prefix1) || s.startsWith(prefix2) || s.startsWith(prefix3) || s.endsWith(suffix1) || s.endsWith(suffix2),
+            () -> randomAlphaOfLengthBetween(10, 12));
+        final String exact3 = randomValueOtherThanMany(
+            s -> s.startsWith(prefix1) || s.startsWith(prefix2) || s.startsWith(prefix3) || s.endsWith(suffix1) || s.endsWith(suffix2),
+            () -> randomAlphaOfLengthBetween(15, 20));
+
+        final StringMatcher matcher = StringMatcher.of(List.of(
+            prefix1 + "*", prefix2 + "?", "/" + prefix3 + "@/", "*" + suffix1, "/@" + suffix2 + "/", exact1, exact2, exact3
+        ));
+
+        assertMatch(matcher, exact1);
+        assertMatch(matcher, exact2);
+        assertMatch(matcher, exact3);
+        assertMatch(matcher, randomAlphaOfLength(3) + suffix1);
+        assertMatch(matcher, randomAlphaOfLength(3) + suffix2);
+        assertMatch(matcher, prefix1 + randomAlphaOfLengthBetween(1, 5));
+        assertMatch(matcher, prefix2 + randomAlphaOfLength(1));
+        assertMatch(matcher, prefix3 + randomAlphaOfLengthBetween(1, 5));
+
+        assertNoMatch(matcher, prefix2 + randomAlphaOfLength(2));
+
+        final char[] nonAlpha = "@/#$0123456789()[]{}<>;:%&".toCharArray();
+        assertNoMatch(matcher, randomFrom(nonAlpha) + randomFrom(exact1, prefix1, suffix1) + randomFrom(nonAlpha));
+        assertNoMatch(matcher, randomFrom(nonAlpha) + randomFrom(exact2, prefix2, suffix2) + randomFrom(nonAlpha));
+        assertNoMatch(matcher, randomFrom(nonAlpha) + randomFrom(exact3, prefix3) + randomFrom(nonAlpha));
+    }
+
+    public void testToString() throws Exception {
+        // Replace any '/' characters because they're meaningful at the start, and just removing them all is simpler
+        final String text1 = RandomizedTest.randomUnicodeOfLengthBetween(5, 80).replace('/', '.');
+        final String text2 = RandomizedTest.randomUnicodeOfLength(20).replace('/', '.');
+        final String text3 = RandomizedTest.randomUnicodeOfLength(50).replace('/', '.');
```

Comment:
Nit: why not use the equal method of `randomUnicodeOfLengthBetween` that is available as part of `EsTestCase`? 
Also, if we test unicode for description here, should it also be tested for matching?